soccer/gameplay/evaluation/passing_positioning.py

```diff
@@ -0,0 +1,175 @@
+# @param weights: A tuple of the 4 different weights to apply to the evaulations overall
+#               (space, field_position, shot_chance, kick_proximty)
+# @return Returns a score between 0 and 1 on how good of pass would be
+def eval_singl_point(kick_point, receive_point, ignore_robots, field_weights,
+                     weights):
+
+    if kick_point is None:
+        if main.ball().valid:
+            kick_point = main.ball().pos
+        else:
+            return None
+
+    shotChance = 0
+
+    # Dissallow shooting over midfield
+    if (kick_point.y > constants.Field.Length / 2):
+        # TODO: Replace with KickEval
+        shotChance = evaluation.shooting.eval_shot(receive_point)
+
+    # TODO: Replace with KickEval
+    passChance = evaluation.passing.eval_pass(kick_point, receive_point)
+
+    space = evaluation.field.space_coeff_at_pos(receive_point, ignore_robots)
+    fieldPos = evaluation.field.field_pos_coeff_at_pos(
+        receive_point, field_weights[0], field_weights[1], field_weights[2])
+    distance = math.exp(-1 * (kick_point - receive_point).mag())
+
+    # All of the other scores are based on whether the pass will actually make it to it
+    # Not worth returning a great position if we cant even get a pass there
+    totalChance = passChance * ( weights[0] * (1 - space) + \
+                                 weights[1] * fieldPos + \
+                                 weights[2] * shotChance + \
+                                 weights[3] * (1 - distance) )
+
+    return totalChance / math.fsum(weights)
+
+
+## Finds the best position to pass to
+#
+# @param kick_point: Point that we are passing from
+# @param evaluation_zone: Area to evaluate passes to
+# @param ignore_robots: Robots to ignore when calculating scores
+# @param field_weights: A tuple of the 3 difference weights to apply to field position 
+#               (Centerness, Distance to their goal, Angle off their goal)
+# @param weights: A tuple of the 4 different weights to apply to the evaulations overall (Weights are normalized)
+#               (space, field_position, shot_chance, kick_proximity)
+# @param debug: Displays pass lines and scores onto the field when true
+# @return bestPoint and bestScore in that order
+def eval_best_receive_point(kick_point,
+                            evaluation_zone=None,
+                            ignore_robots=[],
+                            field_weights=(0.1, 3.2, 0.1),
+                            weights=(1, 4, 15, 1),
+                            debug=False):
+
+    if evaluation_zone is None:
+        evaluation_zone = generate_default_rectangle(kick_point)
+
+    points = get_points_from_rect(evaluation_zone, kick_point, 0.6, 0.5)
+
+    if points is None or len(points) == 0:
+        # Nothing can be done
+        return None, 0
+
+    # TODO: Setup to be a list of the top X points
+    bestScore = None
+    bestPointt = None
```

Comment:
There actually is a tool for this kind of thing, `flake8`, but no one uses it because it spits out wayy too many warnings. We can probably put a toned down version of flake8 into our CI system to catch python errors.